front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx

```diff
@@ -252,33 +207,74 @@ export default function AssistantBuilderDataSourceModal({
           <DataSourceResourceSelector
             dataSource={selectedDataSource}
             owner={owner}
-            selectedResources={selectedResources}
-            isSelectAll={isSelectAll}
+            selectedResources={
+              dataSourceConfigurations[selectedDataSource.name]
+                ?.selectedResources || []
+            }
+            isSelectAll={
+              dataSourceConfigurations[selectedDataSource.name]?.isSelectAll ||
+              false
+            }
             onSelectChange={(node, selected) => {
-              setSelectedResources((currentResources) => {
-                const isNodeAlreadySelected = currentResources.some(
-                  (resource) => resource.internalId === node.internalId
-                );
+              setDataSourceConfigurations((currentConfigurations) => {
+                if (currentConfigurations === null) {
+                  // Unreachable
+                  return null;
+                }
+                const oldConfiguration = currentConfigurations[
+                  selectedDataSource.name
+                ] || {
+                  dataSource: selectedDataSource,
+                  selectedResources: [],
+                  isSelectAll: false,
+                };
+
+                const newConfiguration = {
+                  ...oldConfiguration,
+                };
+
                 if (selected) {
-                  if (!isNodeAlreadySelected) {
-                    return [...currentResources, node];
-                  }
+                  newConfiguration.selectedResources = [
+                    ...oldConfiguration.selectedResources,
+                    node,
+                  ];
                 } else {
-                  if (isNodeAlreadySelected) {
-                    return currentResources.filter(
+                  newConfiguration.selectedResources =
+                    oldConfiguration.selectedResources.filter(
                       (resource) => resource.internalId !== node.internalId
                     );
-                  }
                 }
-                return currentResources;
+
+                return {
+                  ...currentConfigurations,
+                  [selectedDataSource.name]: newConfiguration,
+                };
               });
             }}
             toggleSelectAll={() => {
-              const selectAll = !isSelectAll;
-              if (isSelectAll === false) {
-                setSelectedResources([]);
-              }
-              setIsSelectAll(selectAll);
+              setDataSourceConfigurations((currentConfigurations) => {
```

Comment:
Will this allow to unselect a datasource? 
It was really ugly before, we had to selectall/unselect all to remove a datasource.

Reply:
Doesn't change anything compared to now

Reply:
Not sure if that was your question ?

Reply:
discussed IRL!